dfs/upgraderegistry.go

```diff
@@ -49,6 +51,17 @@ func (dfs *DistributedFilesystem) UpgradeRegistry(svcs []service.Service, tenant
 			glog.Warningf("Cannot parse image name %s under service %s (%s)", image, svc.Name, svc.ID)
 			continue
 		}
+		// download image from old registry
+		if registryHost != "" {
+			glog.V(2).Info("Downloading image %s from %s registry", image, registryHost)
+			oldImage := fmt.Sprintf("%s/%s", registryHost, rImage)
+			if err := dfs.docker.PullImage(oldImage); err != nil {
```

Comment:
Yes, and you will set the host of the v1 registry from where the images are being pulled